### PR TITLE
RTS: Details page

### PR DESCRIPTION
**Pull request by mmtr**
Fixes https://github.com/Automattic/dotcom-forge/issues/7812

## Proposed Changes

Creates a new Readymade Template details page within the Pattern Library.

container-flamboyant-hofstadter calypso live_patterns_site-layouts_1

A few notes:
- When going back to patterns, the page is automatically scrolled to the top. This is a known issue, ideally we should scroll to the RTS section again. It's not easy to solve because of [the current enforced scrolling behavior](https://github.com/Automattic/wp-calypso/blob/a900a7934a2be3ef5fe8d78a007e702b820dac38/client/my-sites/patterns/components/pattern-library/index.tsx#L130-L143), but I don't think it's a blocker for the MVP. We can address it later on.
- Current descriptions look empty. We should update them in the API to make them more exciting.
- The "Pick this layout" button requires changes from https://github.com/Automattic/wp-calypso/pull/92031

 

## Why are these changes being made?

To connect the PL with the new RT onboarding flow

## Testing Instructions

- Use the Calypso link below
- Go to `/patterns`
- Scroll down to the RTS section
- Click on a RT
- Make sure the details page is opened correctly
- Make sure it looks good in different viewports
- Make sure the "All about patterns" section at the bottom as a light color scheme
- Make sure the "Copy layout" button copies the template content to your clipboard (you can paste it in the site editor)
- Make sure the "Pick this layout" button triggers the RT onboarding flow implemented in https://github.com/Automattic/wp-calypso/pull/92031

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/my-sites/patterns/components/readymade-template-details/index.tsx [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+	return (
+		<>
+			<section className="readymade-template-details-wrapper">
+				<div className="readymade-template-details-section">
+					<a
+						href={ addLocaleToPathLocaleInFront( '/patterns' ) }
+						className="readymade-template-details-back"
+					>
+						<Icon icon={ arrowLeft } size={ 24 } /> { translate( 'Back to patterns' ) }
```
Comment (dsas) [inline client/my-sites/patterns/components/readymade-template-details/index.tsx]: Unsure about this, and whether people will recognise that "Back to patterns" is a chance to choose a different "site layout", particularly as going back scrolls to the top of the page.

Maybe just saying 'Back' or 'Back to patterns and layouts' would help.

Comment (mmtr) [reply client/my-sites/patterns/components/readymade-template-details/index.tsx]: Changed it to just "Back": https://github.com/Automattic/wp-calypso/pull/92078/commits/ade6a8072baa42508890fcbee7b8c9c1e5b5b638

**Review by dsas — APPROVED** [review]
One minor non-blocking comment is that there's not much space for titles on larger screen sizes, anything over thirteen characters overflows into an ellipses.
Screenshot 2024-06-26 at 11 35 44

**Comment by rcrdortiz** [comment]
When navigating back I go to the top of the page instead of having the RT list in view. No biggie, but it would be great from a UX standpoint to go back to the list.

**Comment by mmtr** [comment]
@rcrdortiz yes, already noted in the PR summary:

> When going back to patterns, the page is automatically scrolled to the top. This is a known issue, ideally we should scroll to the RTS section again. It's not easy to solve because of [the current enforced scrolling behavior](https://github.com/Automattic/wp-calypso/blob/a900a7934a2be3ef5fe8d78a007e702b820dac38/client/my-sites/patterns/components/pattern-library/index.tsx#L130-L143), but I don't think it's a blocker for the MVP. We can address it later on.

**Comment by rcrdortiz** [comment]
> @rcrdortiz yes, already noted in the PR summary:
> 
> > When going back to patterns, the page is automatically scrolled to the top. This is a known issue, ideally we should scroll to the RTS section again. It's not easy to solve because of [the current enforced scrolling behavior](https://github.com/Automattic/wp-calypso/blob/a900a7934a2be3ef5fe8d78a007e702b820dac38/client/my-sites/patterns/components/pattern-library/index.tsx#L130-L143), but I don't think it's a blocker for the MVP. We can address it later on.

Oh, didn't notice. Sorry for the noise. 🚢

**Review by rcrdortiz — APPROVED** [review]
+1

**Comment by mmtr** [comment]
> there's not much space for titles on larger screen sizes

@dsas Yeah, I noted that too, and added the ellipsis overflowing as a fallback, but I agree there must be a better design solution for this. cc @javierarce

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/15396379

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @mmtr for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/15396379).

**Pull request merged**